Sources/SwiftProtobuf/JSONDecoder.swift

```diff
@@ -466,7 +466,7 @@ internal struct JSONDecoder: Decoder {
       value = nil
       return
     }
-    value = try scanner.nextEnumValue() as E
+    value = try scanner.nextEnumValue()
```

Comment:
I think map decoding also comes thru here, so the value will be nil, and the pair won't end up in the map.  We'll need to confirm that is correct vs. using the default value.  (we can correct this on the map handling side since single field might want to say unset in this case).